File: sorina-nedelcu/Exercise/src/util/Property.php
<?php
class Property {

  private $name;
  private $value;

  public function __construct($name, $value) {
    $this::setName($name);
    $this::setValue($value);
  }

  public function setName($name) {
    $this->name = $name;
  }

  public function getName() {
    return $this->name;
  }

  public function setValue($value) {
    $this->value = $value;
  }

  public function getValue() {
    return $this->value;
  }
}

File: claudia-diaconescu/commandParser/commandParser/src/context/Property.java
package context;

/**
 * @author <a href="mailto:c.diaconescu@moodmedia.ro">Claudia Diaconescu</a>
 * @version $Revision:$
 */
public class Property implements Comparable {
  private String name;
  private String value;

  public Property(String name) {
    this.name = name;
  }

  public Property(String name, String value) {
    this.name = name;
    this.value = value;
  }

  /**
   * Gets property name
   * @return property name
   */
  public String getName() {
    return name;
  }

  /**
   * Sets property name
   * @param name
   */
  public void setName(String name) {
    this.name = name;
  }

  /**
   * Gets property value
   * @return property value
   */
  public String getValue() {
    return value;
  }

  /**
   * Sets property value
   * @param value
   */
  public void setValue(String value) {
    this.value = value;
  }

  @Override
  public int compareTo(Object o) {
    if (o == null ) throw new IllegalArgumentException("Parameter can not be null!");
    if (!(o instanceof Property)) throw new IllegalArgumentException("Parameter must be of Property type!");
    return this.getName().compareTo(((Property)o).getName());
  }

}
